mirgecom/boundary.py

```diff
@@ -38,91 +40,264 @@
+                 boundary_gradient_cv_func=None,
+                 # Returns the boundary value for grad(temperature)
+                 boundary_gradient_temperature_func=None,
+                 ):
         """Initialize the PrescribedFluidBoundary and methods."""
         self._bnd_state_func = boundary_state_func
-        self._inviscid_bnd_flux_func = inviscid_boundary_flux_func
-        self._inviscid_div_flux_func = inviscid_facial_flux_func
+        self._inviscid_flux_func = inviscid_flux_func
+        self._temperature_grad_flux_func = temperature_gradient_flux_func
         self._bnd_temperature_func = boundary_temperature_func
+        self._grad_num_flux_func = gradient_numerical_flux_func
+        self._cv_gradient_flux_func = cv_gradient_flux_func
+        self._viscous_flux_func = viscous_flux_func
+        self._bnd_grad_cv_func = boundary_gradient_cv_func
+        self._bnd_grad_temperature_func = boundary_gradient_temperature_func
 
-        if not self._inviscid_bnd_flux_func and not self._bnd_state_func:
+        if not self._inviscid_flux_func and not self._bnd_state_func:
             from warnings import warn
             warn("Using dummy boundary: copies interior solution.", stacklevel=2)
 
-        if not self._inviscid_div_flux_func:
-            self._inviscid_div_flux_func = inviscid_facial_flux
+        if not self._inviscid_flux_func:
+            self._inviscid_flux_func = self._inviscid_flux_for_prescribed_state
+        # if not self._inviscid_num_flux_func:
+        #     self._inviscid_num_flux_func = inviscid_facial_flux
         if not self._bnd_state_func:
-            self._bnd_state_func = self._dummy_state_func
-        if not self._bnd_temperature_func:
-            self._bnd_temperature_func = self._dummy_temperature_func
-
-    def _dummy_temperature_func(self, temperature_minus, **kwargs):
-        return -temperature_minus
+            self._bnd_state_func = self._identical_state
 
-    def _dummy_state_func(self, state_minus, **kwargs):
-        return state_minus
+        if not self._bnd_temperature_func:
+            self._bnd_temperature_func = self._temperature_for_prescribed_state
+        if not self._grad_num_flux_func:
+            self._grad_num_flux_func = gradient_flux_central
+
+        if not self._cv_gradient_flux_func:
+            self._cv_gradient_flux_func = self._gradient_flux_for_prescribed_cv
+        if not self._temperature_grad_flux_func:
+            self._temperature_grad_flux_func = \
+                self._gradient_flux_for_prescribed_temperature
+
+        if not self._viscous_flux_func:
+            self._viscous_flux_func = self._viscous_flux_for_prescribed_state
+        if not self._bnd_grad_cv_func:
+            self._bnd_grad_cv_func = self._identical_grad_cv
+        if not self._bnd_grad_temperature_func:
+            self._bnd_grad_temperature_func = self._identical_grad_temperature
 
     def _boundary_quantity(self, discr, btag, quantity, **kwargs):
```

Comment:
This takes in arbitary kwargs without checking their validity, inviting typos.

Reply:
([b0a4356](https://github.com/illinois-ceesd/mirgecom/pull/581/commits/b0a4356683cb04e161ef951ceb9d6094241d6992)